src/menus/lists.ts

```diff
@@ -77,66 +77,71 @@ export function getMenuItems (url: string, lang: string, type: string, pageType:
     let err = 0;
     info('Start getting page breadcrumb/siblings', {url: url, lang: lang, method: 'getMenuItems: '.concat(type)});
     let items: MenuEntry[] = [];
-    const m = getCachedMenus();
-    let siteArray: SiteTreeInstance | undefined = m.menus[lang].getMenus();
+    try {
+        const m = getCachedMenus();
+        let siteArray: SiteTreeInstance | undefined = m.menus[lang].getMenus();
+
+        if (siteArray) {
+            const firstSite: { result: { [urlInstance: string]: MenuEntry } | undefined, objectType: string } = siteArray.findItemAndObjectTypeByUrl(url);
+
+            if (firstSite.result) {
+                const restUrl = Object.keys(firstSite.result)[0];
+                info('Page found', {url: restUrl, lang: lang, method: 'getMenuItems: '.concat(type)});
+
+                items = getListFromFirstSite(firstSite.result, restUrl, type, items, siteArray, lang );
+                orphan_pages_counter.labels( {url: url, lang: lang }).set(0);
+            } else {
+                if (firstSite.objectType != 'custom' && firstSite.objectType != 'post') {
+                    error('orphan_page', {url: url, lang: lang});
+                    orphan_pages_counter.labels( {url: url, lang: lang }).set(1);
+                    err ++;
+                }
+                if (pageType == 'post' && type == 'breadcrumb') {
+                    //if the site is not found and we are looking for a post page not attached to the menu,
+                    //we will found the breadcrumb for his site home page and manually add the home post page and the current post page
```

Comment:
I know that you are only moving that section of code around, but please fix the english of this comment.